pkg/sql/sem/builtins/builtins.go

```diff
@@ -5084,6 +5084,44 @@ table's zone configuration this will return NULL.`,
 			Volatility: tree.VolatilityVolatile,
 		},
 	),
+	// Deletes the underlying spans backing a table, only
+	// if the user provides explicit acknowledgement of the
+	// form "I acknowledge this will irrevocably delete all revisions
+	// for table %d"
+	"crdb_internal.force_delete_table_data": makeBuiltin(
+		tree.FunctionProperties{
+			Category: categorySystemRepair,
+		},
+		tree.Overload{
+			Types: tree.ArgTypes{{"id", types.Int},
+				{"accept", types.String}},
+			ReturnType: tree.FixedReturnType(types.Bool),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				id := int64(*args[0].(*tree.DInt))
+				agree := string(*args[1].(*tree.DString))
+				expectedAgree := fmt.Sprint("I acknowledge this will irrevocably delete all revisions for table ", id)
```

Comment:
nit, cc @jordanlewis: why does this string use the word "revisions"? It may be wrong but my interpretation of "revisions" associates it to table _metadata_ i.e. the descriptor instead of table _data_, which is what this is about.

Reply:
On a related note, if this function can only be called on descriptor IDs with no associated descriptors, why is this safety string even required at all? Going even further, couldn't we complement this per-ID function with another diagnostic function that returns the following resultset:
- for each `id` from `keys.MaxReservedDescID` to the value at `codec.DescIDSequenceKey()`:
- if `id` has a descriptor in `system.descriptor`, continue;
- execute a scan request over the table range for `id` with `max_span_request_keys = 1` or something like that;
- if the response is empty, continue;
- add `id` to the resultset.
If this is sane, consider adding this diagnostic function to the doctor suite?

Reply:
How do you feel about a follow on issue with this? I'll add a second diagnostic function, where the result set will be descriptors with table data inside them. @postamar @ajwerner